File: src/main/java/com/jacuum/engine/SessionView.java
package com.jacuum.engine;

public record SessionView(
    String id,
    RunStatus status,
    int robotX,
    int robotY,
    int score,
    int totalCleaned,
    int iterationsUsed,
    int iterationsAvailable,
    int totalFloor,
    FinishReason finishReason
) {}

File: src/main/kotlin/com/jacuum/engine/SessionView.kt
package com.jacuum.engine

data class SessionView(
    val id: String,
    val status: RunStatus,
    val robotX: Int,
    val robotY: Int,
    val score: Int,
    val totalCleaned: Int,
    val iterationsUsed: Int,
    val iterationsAvailable: Int,
    val totalFloor: Int,
    val finishReason: FinishReason?
)
